src/Writers/CSharpWriter/CollectionGetByIdIndexer.cs

```diff
@@ -14,7 +14,7 @@ public class CollectionGetByIdIndexer : Indexer
 
         public CollectionGetByIdIndexer(OdcmClass odcmClass)
         {
-            var keyProperties = odcmClass.Key;
+            var keyProperties = ((OdcmEntityClass)odcmClass).Key;
 
```

Comment:
Since this indexer is only applicable to Entity Classes, why not make the parameter type OdcmEntityClass?